test/e2e/deployment.go

```diff
@@ -458,7 +458,7 @@ func testRolloverDeployment(f *framework.Framework) {
 	// Create a deployment to delete nginx pods and instead bring up redis-slave pods.
 	// We use a nonexistent image here, so that we make sure it won't finish
 	deploymentName, deploymentImageName := "test-rollover-deployment", "redis-slave"
-	deploymentReplicas := int32(4)
+	deploymentReplicas := int32(3)
```

Comment:
- RS-1 has one healthy replica
- D with maxUnavailable=0 creates RS-2 withone unhealthy replica. It shouldn't scale down RS-1 otherwise it violates availability.
- D is updated. RS-3 is scaled up to one. POD-3 becomes healthy, RS-2 and RS-1 are scaled down

Seems doable.